Assignment5/Assignment5.js

```diff
@@ -1,38 +1,27 @@
 const fs = require('fs');
-var read = function(readFile) {
-  return new Promise((resolve,reject) => {
-    fs.readFile(readFile, function(error, dataFromFile) {
-      if (error)
-        reject(error);
-      resolve(dataFromFile);
-      console.log("------------------------------------------------------------");
-      console.log(dataFromFile.toString());
-    });
-  });
-};
-var append = function(appendFile, dataFromFile) {
-  return new Promise((resolve,reject) => {
-    fs.appendFile(appendFile, dataFromFile.toString(), function(error) {
-      if (error)
-        reject(error);
-      resolve();
-    });
-  });
-};
-var write = function(writeFile, dataFromFile) {
-  return new Promise((resolve,reject) => {
-    fs.writeFile(writeFile, dataFromFile, function(error) {
-      if (error)
-        reject(error);
-      resolve();
-    });
-  });
-};
-
-read('file1.txt')
- .then(result => { return append('file2.txt', result) })
- .then(result => { return read('file2.txt') })
- .then(result => { return write('file3.txt', result) })
- .then(result => { return read('file3.txt') })
- .catch(error => console.log(error))
 
+var makePromise = (fileName, fileOperation, tempData) => {
+  return new Promise((resolve, reject) => {
+    fs[fileOperation](fileName, tempData, (error, dataFromFile) => {
+      error ? reject(error) : resolve(dataFromFile);
+    })
+  })
+}
+var print = (string) => {
+  console.log(string);
+}
+makePromise('file1.txt', "readFile", "")
+ .then(result => { print("---------------------------------------------");
+                   print(result.toString());
+                   return makePromise('file2.txt', "appendFile", result) 
+                 })
+ .then(result => { return makePromise('file2.txt', "readFile", "") })
```

Comment:
@aarushi121 Here we don't need to write the return statement for single line. we can use this `.then(result =>  makePromise('file2.txt', "readFile", ""))`